packages/blockchain/bsc/src/lib/services/bsc.offchain.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  AccountService,
+  ApiServices,
+  Currency,
+  TransferBsc,
+  TransferBscKMS,
+  VirtualCurrencyService,
+  WithdrawalService,
+} from '@tatumio/api-client'
+import {
+  abstractBlockchainOffchain,
+  FromPrivateKeyOrSignatureIdOrMnemonic,
+} from '@tatumio/shared-blockchain-abstract'
+import { evmBasedUtils, EvmBasedWeb3 } from '@tatumio/shared-blockchain-evm-based'
+import { Blockchain } from '@tatumio/shared-core'
+import BigNumber from 'bignumber.js'
+import { bscTxService } from './bsc.tx'
+
+type TransferBscOffchain = FromPrivateKeyOrSignatureIdOrMnemonic<TransferBsc>
+type OffchainResponse = { id?: string; txId?: string; completed?: boolean } | void
+
+const sendBscOffchainTransaction = async (
+  body: TransferBscOffchain,
+  web3: EvmBasedWeb3,
+): Promise<OffchainResponse> => {
+  const txService = bscTxService({ blockchain: Blockchain.BSC, web3 })
+  const { mnemonic, index, privateKey, gasLimit, gasPrice, nonce, ...withdrawal } = body as any
+  const { amount, address } = withdrawal
+  let fromPrivKey: string
+  let txData: any
+
+  if (body.mnemonic && body.index !== undefined) {
+    fromPrivKey = (await ApiServices.blockchain.bsc.bscGenerateAddressPrivateKey({
```

Comment:
never generate / send private keys to API inside SDK. you can use wallet feature from the SDK